src/ResourceManagement/Storage/StorageAccountImpl.cs

```diff
@@ -0,0 +1,376 @@
+        }
+
+        ///GENMHASH:99D5BF64EA8AA0E287C9B6F77AAD6FC4:220D4662AAC7DF3BEFAF2B253278E85C
+        public ProvisioningState ProvisioningState()
+        {
+            // TODO: In the next major version upgrade of SDK change return type to nullable
+            //       returning the default enum value when server returns null is not expected
+            return Inner.ProvisioningState.HasValue ? Inner.ProvisioningState.Value : default(ProvisioningState);
+        }
+
+        ///GENMHASH:55B42F94DE82BCEA956E2090896DCEA5:93EB9B10265EA252D67B32BBB824A1F6
+        public PublicEndpoints EndPoints()
+        {
+            return new PublicEndpoints(Inner.PrimaryEndpoints, Inner.SecondaryEndpoints);
+        }
+
+        ///GENMHASH:C7B3D3963622074C2FD00EB9A2E0FE72:84CE69ECBC3078017A0DFFC90EF4E3D9
+        public Encryption Encryption()
+        {
+            return Inner.Encryption;
+        }
+
+        ///GENMHASH:E3CB6E557BDC02538C5A6963772F3FEF:349706F064F75B85DB63EF492656563E
+        public StorageAccountEncryptionKeySource EncryptionKeySource()
+        {
+            if (this.Inner.Encryption == null || Microsoft.Azure.Management.Storage.Fluent.Models.Encryption.KeySource == null) {
+                return null;
+            }
+            return StorageAccountEncryptionKeySource.Parse(Microsoft.Azure.Management.Storage.Fluent.Models.Encryption.KeySource);
+        }
+
+        ///GENMHASH:26FB96D5CAED2DAC6A25B7684BA6EA62:EA8D98221847758B674F32A5F6BA8D4E
+        public IReadOnlyDictionary<StorageService, IStorageAccountEncryptionStatus> EncryptionStatuses()
+        {
+            var statuses = new Dictionary<StorageService, IStorageAccountEncryptionStatus>();
+            if (this.Inner.Encryption != null && this.Inner.Encryption.Services != null)
+            {
+                // Status of blob service
+                //
+                // Status for other service needs to be added as storage starts supporting it
+                statuses.Add(StorageService.Blob, new BlobServiceEncryptionStatusImpl(this.Inner.Encryption.Services));
+            }
+            else
+            {
+                statuses.Add(StorageService.Blob, new BlobServiceEncryptionStatusImpl(new EncryptionServices()));
+            }
+            return statuses;
+        }
+
+        ///GENMHASH:F740873A801629829EA1C3C98F4FDDC4:ACAFFE3955CCFBD0C2BC6D268AECA2BA
+        public AccessTier AccessTier()
+        {
+            // TODO: In the next major version upgrade of SDK change return type to nullable
+            //       returning the default enum value when server returns null is not expected
+            return Inner.AccessTier.HasValue ? Inner.AccessTier.Value : default(AccessTier);
+        }
+
+        ///GENMHASH:E4DFA7EA15F8324FB60C810D0C96D2FF:1BCD5CF569F11AB6F798D4F3A5BFC786
+        public IReadOnlyList<Models.StorageAccountKey> GetKeys()
+        {
+            return GetKeysAsync().ConfigureAwait(false).GetAwaiter().GetResult();
```

Comment:
Please change all ".ConfigureAwait(false).GetAwaiter().GetResult();" to Extensions.Synchronize(() => <your ASYNC method call>)